### PR TITLE
API: Delegation | verify analytics and headers

### PR DESCRIPTION
**Pull request by Sotatek-GiangNguyen7**

### CODE REVIEW

**Comment by Sotatek-GiangNguyen7** [comment]
https://sotatek-davidnguyen4.github.io/cardano-api-tests/48/#suites
image

File: src/test/java/tests/delegation_controller/PoolDetailAnalyticTests.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    @Test(description = "verify that get data uSsuccessfully from params 'poolViews' ", groups={"delegation", "delegation-detail-analytics"}, dataProvider = "paramInvalidPoolView")
+    public void verifyGetDataFromPoolDetailAnalyticsUnsuccessfully(Object poolView){
+        Map<String, Object> param = new CreateParameters().withPoolView(poolView).getParamsMap();
+
+        ErrorResponse errorResponse = (ErrorResponse) delegationPoolDetailAnalyticSteps
+                .getDataForPoolDetailAnalytics(param)
+                .validateStatusCode(HttpURLConnection.HTTP_BAD_REQUEST)
+                .saveResponseObject(ErrorResponse.class);
+        baseSteps.then_verifyErrorResponse(HttpURLConnection.HTTP_BAD_REQUEST, errorResponse.getErrorMessage(), errorResponse.getErrorCode());
```
Comment (ghost) [inline src/test/java/tests/delegation_controller/PoolDetailAnalyticTests.java]: should be delegationPoolDetailAnalyticSteps, delegationPoolDetailAnalyticSteps should extends baseSteps

Comment (ghost) [inline src/test/java/tests/delegation_controller/PoolDetailAnalyticTests.java]: should verify status code in only one place. If verify at line 37, you create new method verifyErrorResponse(errorCode, errorMessage). Or you remove validateStatusCode() at line 37

**Comment by Sotatek-GiangNguyen7** [comment]
https://sotatek-davidnguyen4.github.io/cardano-api-tests/52/#suites
image

File: src/test/java/tests/delegation_controller/PoolDetailAnalyticTests.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import sun.net.www.protocol.http.HttpURLConnection;
+import util.CreateParameters;
+
+import java.util.Map;
+
+@Epic("cardano")
+@Feature("api-delegation")
+public class PoolDetailAnalyticTests extends BaseTest {
+    DelegationPoolDetailAnalyticSteps delegationPoolDetailAnalyticSteps = new DelegationPoolDetailAnalyticSteps();
+    BaseSteps baseSteps = new BaseSteps();
```
Comment (ghost) [inline src/test/java/tests/delegation_controller/PoolDetailAnalyticTests.java]: remove it

**Review by ghost — APPROVED** [review]

**Pull request merged**